src/ansi-c/c_typecheck_code.cpp

```diff
@@ -1123,6 +1196,22 @@ void c_typecheck_baset::typecheck_spec_assigns(exprt::operandst &targets)
   std::swap(targets, tmp);
 }
 
+void c_typecheck_baset::typecheck_spec_frees(exprt::operandst &targets)
+{
+  const std::function<void(exprt &)> typecheck_target = [&](exprt &target) {
+    typecheck_spec_frees_target(target);
+  };
+  typecheck_conditional_targets(targets, typecheck_target, "frees");
```

Comment:
Should _"frees"_  and _"assigns"_ be a macro?

Reply:
what is wrong with std::function ?